lxd/migrate.go

```diff
@@ -72,7 +72,34 @@ func (c *migrationFields) recv(m proto.Message) error {
 		return fmt.Errorf("Control connection not initialized: %w", err)
 	}
 
-	return migration.ProtoRecv(conn, m)
+	buf, err := migration.ReadBytesFromConnection(conn)
+	if err != nil {
+		return err
+	}
+
+	// If we are expecting a MigrationHeader, first check if we received a MigrationControl
+	// which may contain a message indicating that we need to stop.
+	_, ok := m.(*migration.MigrationHeader)
+	if ok {
+		ctl := &migration.MigrationControl{}
+
+		// Enforce required fields present.
```

Comment:
Which part of the code does this? AllowPartialFalse? If so can you move the comment to next to the field?

Reply:
Yes, fixed!